libs/datamodel/core/src/transform/ast_to_dml/db/attributes.rs

```diff
@@ -360,6 +356,44 @@ fn primary_key_constraint_name<'ast>(
     db_name
 }
 
+fn default_value_constraint_name<'ast>(
+    args: &mut Arguments<'ast>,
+    ast_model: &'ast Model,
+    model_data: &mut ModelData<'ast>,
+    ast_field: &'ast ast::Field,
+    ctx: &mut Context<'ast>,
+) -> Option<String> {
+    if !ctx.db.preview_features.contains(NamedConstraints) {
+        return None;
+    }
+
+    let db_name = match args.optional_arg("map").map(|name| name.as_str()) {
+        Some(Ok("")) => error_on_empty_string_cow(args, ctx),
+        Some(Ok(name)) => Some(name.into()),
+        Some(Err(err)) => push_error_cow(err, ctx),
+        None if ctx.db.active_connector().supports_named_default_values() => {
+            let generated_name = ConstraintNames::default_name(
+                model_data.mapped_name.unwrap_or(&ast_model.name.name),
+                ast_field.name(),
```

Comment:
I think this needs to take the mapped name of the field in order to be sure that it matches the column name.

Reply:
Probably best to add a test using @@map and @map and @default together

Reply:
Done in https://github.com/prisma/prisma-engines/pull/2156/commits/21942920d93324704f11e7a7380931a6be08e376